src/coreclr/jit/optimizer.cpp

```diff
@@ -468,27 +464,26 @@ void Compiler::optUpdateLoopsBeforeRemoveBlock(BasicBlock* block, bool skipUnmar
             loop.lpExit = nullptr;
         }
 
-        /* If this points to the actual entry in the loop
-         * then the whole loop may become unreachable */
+        // If `block` flows to the loop entry then the whole loop will become unreachable if it is the
+        // only non-loop predecessor.
 
         switch (block->bbJumpKind)
         {
             case BBJ_NONE:
-            case BBJ_COND:
```

Comment:
The code here does the same thing, but I created separate code for BBJ_NONE/BBJ_COND/BBJ_ALWAYS instead of the confusing flow of BBJ_COND re-using both BBJ_NONE and BBJ_ALWAYS paths using "fall-through".